IdentityCore/src/controllers/MSIDSilentController.m

```diff
@@ -143,19 +143,26 @@ - (void)acquireTokenWithRequest:(MSIDSilentTokenRequest *)request
         self.currentRequest = nil;
         MSIDRequestCompletionBlock completionBlockWrapper = ^(MSIDTokenResult *fallResult, NSError *fallError)
         {
-            // We don't have any meaningful information from fallback controller (edge case of SSO error) so we use the local controller result earlier
-            if (!fallResult && (fallError.code == MSIDErrorSSOExtensionUnexpectedError))
+            // Only return fallback when there is a valid result, otherwise, return error from main controller
+            // (!result && !error) is a special case when using local silent controller and to skip local refreh token (broker first flow). In this case, the main controller is the 1st fallback controller, and should return the error
+            if (fallResult || (!result && !error))
             {
-                completionBlock(result, error);
+                completionBlock(fallResult, fallError);
             }
             else
             {
-                completionBlock(fallResult, fallError);
+                completionBlock(result, error);
```

Comment:
why do we call fallback if we already have result?

Reply:
In this case, the result is nil, the valid part is error only. I can send `nil` specifically